src/labkey/Domain.ts

```diff
@@ -193,4 +193,64 @@ export function save(config: SaveDomainOptions): void {
             domainId: options.domainId
         }
     });
+}
+
+interface ListDomainsParams {
+    domainKinds?: Array<string>
+    includeFields?: boolean
+    includeProjectAndShared?: boolean
+}
+
+export interface ListDomainsOptions extends ListDomainsParams {
+    containerPath?: string
+    failure?: (error?: any) => any
+    success?: (response: {data?: Array<DomainDesign>}) => any
+}
+
+export function listDomains(config: ListDomainsOptions): void {
+    const params: ListDomainsParams = { };
+
+    if (config.domainKinds)
+        params.domainKinds = config.domainKinds;
+
+    if (config.includeFields !== undefined)
+        params.includeFields = config.includeFields;
+
+    if (config.includeProjectAndShared !== undefined)
+        params.includeProjectAndShared = config.includeProjectAndShared;
+
+    request({
+        url: buildURL('property', 'listDomains.api', config.containerPath, params),
+        method: 'GET',
+        success: getCallbackWrapper(config.success),
+        failure: getCallbackWrapper(config.failure, this, true),
+    });
+}
+
+interface GetPropertiesParams {
+    propertyIds?: Array<number>
+    propertyURIs?: Array<string>
+}
+
+export interface GetPropertiesOptions extends GetPropertiesParams{
+    containerPath?: string
+    failure?: (error?: any) => any
+    success?: (response: {data?: Array<any /*DomainField*/>}) => any
```

Comment:
Is this supposed to be DomainField?

Reply:
Yes, it is a DomainField, but labkey-api-js doesn't (yet) have the DomainField type defined in our components package.